lib/remap-functions/src/merge.rs

```diff
@@ -121,88 +112,39 @@ mod tests {
-                    "parent1":
-                        map![ "key1": "val1",
-                              "child": map! [ "grandchild1": "val1" ] ],
-                    "parent2":
-                        map![ "key2": "val2",
-                              "child": map! [ "grandchild2": "val2" ] ]
-                ],
-                map![
-                    "parent1":
-                        map![ "key1": "val1",
-                              "key2": "val2",
-                              "child": map! [ "grandchild1": "val1",
-                                              "grandchild2": "val2" ] ],
-                    "parent2":
-                        map![ "key2": "val2",
-                              "child": map! [ "grandchild2": "val2" ] ]
-                ],
-                MergeFn::new(
-                    Path::from("parent1"),
-                    Box::new(Path::from("parent2")),
-                    Some(Box::new(Literal::from(Value::Boolean(true)))),
-                ),
-                Ok(Value::Null),
-            ),
-        ];
-
-        let mut state = state::Program::default();
-        for (input_event, exp_event, func, exp_result) in cases {
-            let mut input_event = Value::Map(input_event);
-            let exp_event = Value::Map(exp_event);
-
-            let got = func
-                .execute(&mut state, &mut input_event)
-                .map_err(|e| format!("{:#}", anyhow::anyhow!(e)));
-
-            assert_eq!(input_event, exp_event);
-            assert_eq!(got, exp_result);
+    test_function! [
+        merge => Merge;
+
+        simple {
+            args: func_args![ to: map![ "key1": "val1" ],
+                              from: map![ "key2": "val2" ]
+            ],
+            want: Ok(map![ "key1": "val1",
+                            "key2": "val2" ])
         }
-    }
+
+        shallow {
+            args: func_args![ to: map![ "key1": "val1",
+                                         "child": map! [ "grandchild1": "val1" ] ],
+                              from: map![ "key2": "val2",
+                                           "child": map! [ "grandchild2": "val2" ] ]
+            ],
+            want: Ok(map![ "key1": "val1",
+                                    "key2": "val2",
+                                    "child": map! [ "grandchild2": "val2" ] ] )
+        }
+
+        deep {
+            args: func_args![to: map![ "key1": "val1",
+                                        "child": map! [ "grandchild1": "val1" ] ],
+                             from: map![ "key2": "val2",
+                                          "child": map! [ "grandchild2": "val2" ] ],
+                             deep: true
+            ],
+            want: Ok( map![ "key1": "val1",
+                             "key2": "val2",
+                             "child": map! [ "grandchild1": "val1",
+                                              "grandchild2": "val2" ] ])
+        }
+    ];
```

Comment:
Everything within the `test_function!` looks quite weirdly formatted (even the number of spaces for the indentation does not match). Does `cargo fmt` bail in macros?

Reply:
Yeah it does, unfortunately. It's something I really dislike as well 😭

It _sometimes_ works, but it seems to depend on the amount of code inside the macro, not sure how that works.

Reply:
> Does cargo fmt bail in macros?

Yes, and even more annoyingly Emacs always wants to auto-indent the second line with an extra space for some reason, so if I'm not extra cautious these issues creep in. I'll fix it.